src/overlays/actors/ovl_Bg_Ice_Turara/z_bg_ice_turara.c

```diff
@@ -27,23 +41,164 @@ const ActorInit Bg_Ice_Turara_InitVars = {
+        this->shiverTimer--;
+    }
+    if (!(this->shiverTimer % 4)) {
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_ICE_SWING);
+    }
+    if (this->shiverTimer == 0) {
+        this->dyna.actor.posRot.pos.x = this->dyna.actor.initPosRot.pos.x;
+        this->dyna.actor.posRot.pos.z = this->dyna.actor.initPosRot.pos.z;
+        Collider_CylinderUpdate(&this->dyna.actor, &this->collider);
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        func_8003EBF8(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+        this->actionFunc = func_80892424;
+        return;
+    }
+    sp28 = Math_Rand_ZeroOne();
+    phi_v0_2 = (Math_Rand_ZeroOne() < 0.5f ? -1 : 1);
+    this->dyna.actor.posRot.pos.x = (phi_v0_2 * ((0.5f * sp28) + 0.5f)) + this->dyna.actor.initPosRot.pos.x;
+    sp28 = Math_Rand_ZeroOne();
+    phi_v0_3 = (Math_Rand_ZeroOne() < 0.5f ? -1 : 1);
+    this->dyna.actor.posRot.pos.z = (phi_v0_3 * ((0.5f * sp28) + 0.5f)) + this->dyna.actor.initPosRot.pos.z;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Turara/func_80892040.s")
+void func_80892424(BgIceTurara* this, GlobalContext* globalCtx) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Turara/func_80892220.s")
+    while (true) {
+        if ((this->collider.base.atFlags & 2) || (this->dyna.actor.bgCheckFlags & 1)) {
+            this->collider.base.atFlags &= 0xFFFD;
+            this->dyna.actor.bgCheckFlags &= 0xFFFE;
+            if (this->dyna.actor.posRot.pos.y < this->dyna.actor.groundY) {
+                this->dyna.actor.posRot.pos.y = this->dyna.actor.groundY;
+            }
+            BgIceTurara_Break(this, globalCtx, 40.0f);
+            if (this->dyna.actor.params != 2) {
+            } else {
+                this->dyna.actor.posRot.pos.y = this->dyna.actor.initPosRot.pos.y + 120.0f;
+                func_8003EC50(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+                this->actionFunc = func_80892574;
+                return;
+            }
+            Actor_Kill(&this->dyna.actor);
+            return;
+        } else
+            break;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Turara/func_80892280.s")
+    Actor_MoveForward(&this->dyna.actor);
+    this->dyna.actor.posRot.pos.y += 40.0f;
+    func_8002E4B4(globalCtx, &this->dyna.actor, 0.0f, 0.0f, 0.0f, 4);
+    this->dyna.actor.posRot.pos.y -= 40.0f;
+    Collider_CylinderUpdate(&this->dyna.actor, &this->collider);
+    CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Turara/func_808922B8.s")
+void func_80892574(BgIceTurara* this, GlobalContext* globalCtx) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Turara/func_80892424.s")
+    if (Math_ApproxF(&this->dyna.actor.posRot.pos.y, this->dyna.actor.initPosRot.pos.y, 1.0f) != 0) {
```

Comment:
```suggestion
    if (Math_ApproxF(&this->dyna.actor.posRot.pos.y, this->dyna.actor.initPosRot.pos.y, 1.0f)) {
```
Math_ApproxF returns a boolean value.